libdevcore/CommonData.cpp

```diff
@@ -155,3 +158,14 @@ bool dev::isValidDecimal(string const& _string)
 		return false;
 	return true;
 }
+
+// Returns a quoted string if all characters are printable ASCII chars,
+// or its hex representation otherwise.
+std::string dev::formatAsStringOrNumber(std::string const& _value)
+{
+	for (auto const& c: _value)
+		if (!isprint(c))
```

Comment:
This may be locale-dependent...

Reply:
I'm not sure, its docs state that
` In C++, a locale-specific template version of this function (isprint) exists in header <locale>.`

Reply:
ah, that doesn't mean the negation of your statement

Reply:
What's the rule for it to be safely printable?

Reply:
Oh we should also exclude `"` of course (or escape it).

Maybe go with
```
     digits (0123456789)
    uppercase letters (ABCDEFGHIJKLMNOPQRSTUVWXYZ)
    lowercase letters (abcdefghijklmnopqrstuvwxyz)
    punctuation characters excluding double quote (!#$%&'()*+,-./:;<=>?@[\]^_`{|}~)
    space ( ) 
```

Reply:
Perhaps we are actually safe if `locale` is not included: http://www.cplusplus.com/reference/cctype/isprint/

> For the standard ASCII character set (used by the "C" locale), printing characters are all with an ASCII code greater than 0x1f (US), except 0x7f (DEL).
>
>  In C++, a locale-specific template version of this function (isprint) exists in header <locale>.